p2p/libp2p/libp2pConnectionMonitor.go

```diff
@@ -8,23 +8,21 @@ import (
 	"github.com/multiformats/go-multiaddr"
 )
 
-// ThresholdMinimumConnectedPeers if the number of connected peers drop under this value, for each disconnecting
-// peer, a trigger to reconnect to initial peers is done
-var ThresholdMinimumConnectedPeers = 3
-
 // DurationBetweenReconnectAttempts is used as to not call reconnecter.ReconnectToNetwork() to often
 // when there are a lot of peers disconnecting and reconnection to initial nodes succeed
 var DurationBetweenReconnectAttempts = time.Second * 5
 
 type libp2pConnectionMonitor struct {
-	chDoReconnect chan struct{}
-	reconnecter   p2p.Reconnecter
+	chDoReconnect              chan struct{}
+	reconnecter                p2p.Reconnecter
+	thresholdMinConnectedPeers int
 }
 
-func newLibp2pConnectionMonitor(reconnecter p2p.Reconnecter) *libp2pConnectionMonitor {
+func newLibp2pConnectionMonitor(reconnecter p2p.Reconnecter, thresholdMinConnectedPeers int) *libp2pConnectionMonitor {
 	cm := &libp2pConnectionMonitor{
-		reconnecter:   reconnecter,
-		chDoReconnect: make(chan struct{}, 0),
+		reconnecter:                reconnecter,
+		chDoReconnect:              make(chan struct{}, 0),
+		thresholdMinConnectedPeers: thresholdMinConnectedPeers,
```

Comment:
You can add a check for this new field (not negative, stuff like that) and return an error if the value is wrong. Also, add a test for this case

Reply:
+1

Reply:
Done